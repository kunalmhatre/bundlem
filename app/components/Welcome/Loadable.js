/**
 *
 * Asynchronously loads the component for Welcome
 *
 */

import loadable from 'loadable-components';

export default loadable(() => import('./index'));
