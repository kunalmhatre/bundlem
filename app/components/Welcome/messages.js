/*
 * Welcome Messages
 *
 * This contains all the text for the Welcome component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Welcome';

export default defineMessages({
  bundlem: {
    id: `${scope}.bundlem`,
    defaultMessage: 'Bundlem',
  },
  headLine: {
    id: `${scope}.headLine`,
    defaultMessage: 'Create and share your bundle of online resources',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: 'An easy way to bundle, store, and share your online resources like links of articles, videos or documents.',
  },
  createButton: {
    id: `${scope}.createButton`,
    defaultMessage: 'CREATE',
  },
  searchButton: {
    id: `${scope}.searchButton`,
    defaultMessage: 'SEARCH',
  },
  featuredBundles: {
    id: `${scope}.featuredBundles`,
    defaultMessage: 'FEATURED BUNDLES',
  },
  firstFeaturedBundleTitle: {
    id: `${scope}.firstFeaturedBundleTitle`,
    defaultMessage: 'UX - Few amazing and useful resources to build better products and to make your users satisfied',
  },
  secondFeaturedBundleTitle: {
    id: `${scope}.secondFeaturedBundleTitle`,
    defaultMessage: 'Songs which might make you feel good',
  },
});
