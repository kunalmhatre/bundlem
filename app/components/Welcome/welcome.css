.welcome {
    margin-bottom: 2rem;
}

.welcome-logo {
    margin-top: 3rem;
    padding: 1rem 0 1rem 0;
    font-size: 3rem;
    font-weight: 700;
}
.welcome-headline {
    padding: 0.5rem 0 0.5rem 0;
    font-size: 1.3rem;
    font-weight: 700;
}
.welcome-description {
    opacity: 0.7;
    font-size: 0.9rem;
    font-weight: 700;
}
.welcome-buttons {
    padding: 1.3rem 0 1.3rem 0;
}
.welcome-create-button {
    margin-right: 1rem;
}
.welcome-featured-bundles {
    padding: 0.9rem 0 0.9rem 0;
    font-size: 1.1rem;
    font-weight: 700;
    letter-spacing: 0.1rem;
}

@media only screen and (max-width: 767px) {
    .welcome-featured-bundles-lg {
        display: none;
    }
    .welcome-featured-bundles-sm {
        .ant-row {
            margin-bottom: 1rem;
        }
    }
}

@media only screen and (min-width: 768px) {
    .welcome-featured-bundles-sm {
        display: none;
    }
}