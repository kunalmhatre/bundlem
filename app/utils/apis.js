/**
 * REST APIs
 */

export const domain = 'https://bundlem.in';
export const nodeAppDomain = 'https://bundlem.herokuapp.com';

export const verifyTokenAPI = `${nodeAppDomain}/verify`;
export const publishBundleAPI = `${nodeAppDomain}/create`;
export const getBundleAPI = `${nodeAppDomain}/bundle`;
