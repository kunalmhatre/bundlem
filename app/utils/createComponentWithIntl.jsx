/**
 * Renders component with react-intl support for snapshot testing
 * Source: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-2
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) => renderer.create(
  <IntlProvider {...props}>
    {children}
  </IntlProvider>,
);

export default createComponentWithIntl;
