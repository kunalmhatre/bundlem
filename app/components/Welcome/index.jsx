import React from 'react';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import { Row, Col } from 'antd';
import { withRouter } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import { compose } from 'redux';

import PageLayout from '../PageLayout';
import Button from '../Button';
import BundleCard from '../BundleCard';
import messages from './messages';
import './welcome.css';

const propTypes = {
  intl: intlShape.isRequired,
};

/**
 * Welcome component
 */
function Welcome({
  /* eslint-disable react/prop-types */
  history,
  intl, // react-intl prop
}) {
  const helmetTitleFM = intl.formatMessage(messages.helmetTitle);
  const descriptionFM = intl.formatMessage(messages.description);
  const responsiveDescription = {
    xs: 24,
    sm: 22,
    md: 18,
    xl: 10,
    xxl: 8,
  };
  const responsiveFeaturedBundle = {
    md: 10,
    xl: 8,
    xxl: 6,
  };

  return (
    <React.Fragment>
      <Helmet>
        <title>{helmetTitleFM}</title>
        <meta name="description" content={descriptionFM} />
      </Helmet>
      <PageLayout setHeader={false}>
        <div className="welcome">
          <Row>
            <Col className="welcome-logo">
              <FormattedMessage {...messages.bundlem} />
            </Col>
          </Row>
          <Row>
            <Col className="welcome-headline">
              <FormattedMessage {...messages.headLine} />
            </Col>
          </Row>
          <Row>
            <Col
              className="welcome-description"
              {...responsiveDescription}
            >
              <FormattedMessage {...messages.description} />
            </Col>
          </Row>
          <Row>
            <Col className="welcome-buttons">
              <Button
                className="welcome-create-button"
                text={messages.createButton}
                onClick={() => history.push('/create')}
              />
              <Button
                text={messages.searchButton}
                highlight={false}
                onClick={() => history.push('/bundle/search')}
              />
            </Col>
          </Row>
          <hr />
          <Row>
            <Col>
              <div className="welcome-featured-bundles">
                <FormattedMessage {...messages.featuredBundles} />
              </div>
              <div className="welcome-featured-bundles-lg">
                <Row>
                  <Col {...responsiveFeaturedBundle}>
                    <BundleCard
                      bundleID={57}
                      bundleTitle={messages.firstFeaturedBundleTitle.defaultMessage}
                      resourcesCount={7}
                    />
                  </Col>
                  <Col {...responsiveFeaturedBundle}>
                    <BundleCard
                      bundleID={60}
                      bundleTitle={messages.secondFeaturedBundleTitle.defaultMessage}
                      resourcesCount={6}
                    />
                  </Col>
                </Row>
              </div>
              <div className="welcome-featured-bundles-sm">
                <Row>
                  <Col>
                    <BundleCard
                      bundleID={57}
                      bundleTitle={messages.firstFeaturedBundleTitle.defaultMessage}
                      resourcesCount={7}
                    />
                  </Col>
                </Row>
                <Row>
                  <Col>
                    <BundleCard
                      bundleID={60}
                      bundleTitle={messages.secondFeaturedBundleTitle.defaultMessage}
                      resourcesCount={6}
                    />
                  </Col>
                </Row>
              </div>
            </Col>
          </Row>
        </div>
      </PageLayout>
    </React.Fragment>
  );
}

Welcome.propTypes = propTypes;

export default compose(
  React.memo,
  withRouter,
  injectIntl,
)(Welcome);

const WelcomeComponent = injectIntl(Welcome);

export { WelcomeComponent };
