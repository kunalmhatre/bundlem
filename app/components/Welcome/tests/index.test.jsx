import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import createComponentWithIntl from '../../../utils/createComponentWithIntl';
import { mountWithIntl } from '../../../utils/reactIntlHelperFunction';
import Welcome, { WelcomeComponent } from '../index';

describe('<Welcome />', () => {
  it('renders old snapshot', () => {
    expect(
      createComponentWithIntl(
        <BrowserRouter>
          <Welcome />
        </BrowserRouter>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  it('CREATE button redirects to /create', () => {
    const history = {
      push: jest.fn(pathname => pathname),
    };

    mountWithIntl(
      <WelcomeComponent history={history} />,
    ).find('button')
      .at(0)
      .simulate('click');

    expect(history.push).toHaveReturnedWith('/create');
  });

  it('SEARCH button redirects to /bundle/search', () => {
    const history = {
      push: jest.fn(pathname => pathname),
    };

    mountWithIntl(
      <WelcomeComponent history={history} />,
    ).find('button')
      .at(1)
      .simulate('click');

    expect(history.push).toHaveReturnedWith('/bundle/search');
  });
});
