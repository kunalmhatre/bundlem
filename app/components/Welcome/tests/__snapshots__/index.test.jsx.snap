// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> renders old snapshot 1`] = `
<div
  className="ant-row page-layout theme-blue"
>
  <div
    className="ant-col-22 ant-col-offset-1"
  >
    <div
      className="page-layout-header-content"
    >
      <section
        className="page-layout-content"
      >
        <div
          className="welcome"
        >
          <div
            className="ant-row"
          >
            <div
              className="welcome-logo"
            >
              <span>
                Bundlem
              </span>
            </div>
          </div>
          <div
            className="ant-row"
          >
            <div
              className="welcome-headline"
            >
              <span>
                Create and share your bundle of online resources
              </span>
            </div>
          </div>
          <div
            className="ant-row"
          >
            <div
              className="welcome-description ant-col-xs-24 ant-col-sm-22 ant-col-md-18 ant-col-xl-10 ant-col-xxl-8"
            >
              <span>
                An easy way to bundle, store, and share your online resources like links of articles, videos or documents.
              </span>
            </div>
          </div>
          <div
            className="ant-row"
          >
            <div
              className="welcome-buttons"
            >
              <button
                className="button theme-blue-inverse welcome-create-button"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  CREATE
                </span>
                
              </button>
              <button
                className="button theme-blue"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  SEARCH
                </span>
                
              </button>
            </div>
          </div>
          <hr />
          <div
            className="ant-row"
          >
            <div
              className=""
            >
              <div
                className="welcome-featured-bundles"
              >
                <span>
                  FEATURED BUNDLES
                </span>
              </div>
              <div
                className="welcome-featured-bundles-lg"
              >
                <div
                  className="ant-row"
                >
                  <div
                    className="ant-col-md-10 ant-col-xl-8 ant-col-xxl-6"
                  >
                    <div
                      className="bundle-card theme-blue-inverse"
                    >
                      <div
                        className="bundle-card-title"
                      >
                        <a
                          className="theme-blue-inverse"
                          href="https://bundlem.in/bundle/57"
                        >
                          UX - Few amazing and useful resources to build better products and to make your users satisfied
                        </a>
                      </div>
                      <div
                        className="bundle-card-actions"
                      >
                        <div>
                          <span>
                            7
                          </span>
                          <span>
                             Res.
                          </span>
                        </div>
                        <div>
                          <a
                            href="https://bundlem.in/bundle/57"
                          >
                            <span>
                              view
                            </span>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="ant-col-md-10 ant-col-xl-8 ant-col-xxl-6"
                  >
                    <div
                      className="bundle-card theme-blue-inverse"
                    >
                      <div
                        className="bundle-card-title"
                      >
                        <a
                          className="theme-blue-inverse"
                          href="https://bundlem.in/bundle/60"
                        >
                          Songs which might make you feel good
                        </a>
                      </div>
                      <div
                        className="bundle-card-actions"
                      >
                        <div>
                          <span>
                            6
                          </span>
                          <span>
                             Res.
                          </span>
                        </div>
                        <div>
                          <a
                            href="https://bundlem.in/bundle/60"
                          >
                            <span>
                              view
                            </span>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="welcome-featured-bundles-sm"
              >
                <div
                  className="ant-row"
                >
                  <div
                    className=""
                  >
                    <div
                      className="bundle-card theme-blue-inverse"
                    >
                      <div
                        className="bundle-card-title"
                      >
                        <a
                          className="theme-blue-inverse"
                          href="https://bundlem.in/bundle/57"
                        >
                          UX - Few amazing and useful resources to build better products and to make your users satisfied
                        </a>
                      </div>
                      <div
                        className="bundle-card-actions"
                      >
                        <div>
                          <span>
                            7
                          </span>
                          <span>
                             Res.
                          </span>
                        </div>
                        <div>
                          <a
                            href="https://bundlem.in/bundle/57"
                          >
                            <span>
                              view
                            </span>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-row"
                >
                  <div
                    className=""
                  >
                    <div
                      className="bundle-card theme-blue-inverse"
                    >
                      <div
                        className="bundle-card-title"
                      >
                        <a
                          className="theme-blue-inverse"
                          href="https://bundlem.in/bundle/60"
                        >
                          Songs which might make you feel good
                        </a>
                      </div>
                      <div
                        className="bundle-card-actions"
                      >
                        <div>
                          <span>
                            6
                          </span>
                          <span>
                             Res.
                          </span>
                        </div>
                        <div>
                          <a
                            href="https://bundlem.in/bundle/60"
                          >
                            <span>
                              view
                            </span>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <footer
      className="footer theme-blue"
    >
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/"
      >
        <span>
          source
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/issues/new/choose"
      >
        <span>
          REPORT ISSUE
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/bundlem/"
      >
        <span>
          STORY
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/"
      >
        <span>
          AUTHOR
        </span>
      </a>
    </footer>
  </div>
</div>
`;
